docs/src/main/asciidoc/security-authentication-mechanisms.adoc

```diff
@@ -352,13 +352,145 @@ quarkus.http.auth.policy.role-policy-cert.roles-allowed=user,admin     <2>
 
 Given the preceding configuration, the request is authorized if the client certificate's CN attribute is equal to `alice` or `bob` and forbidden if it is equal to `jdoe`.
 
+[[use-cert-attrs-to-augment-identity]]
 ==== Using certificate attributes to augment SecurityIdentity
 
 You can always register `SecurityIdentityAugmentor` if the automatic <<map-certificate-attributes-to-roles>> option does not suit.
 Custom `SecurityIdentityAugmentor` can check the values of different client certificate attributes and augment the `SecurityIdentity` accordingly.
 
 For more information about customizing `SecurityIdentity`, see the xref:security-customization.adoc#security-identity-customization[Security identity customization] section in the Quarkus "Security tips and tricks" guide.
 
+[[mtls-programmatic-set-up]]
+==== Set up the mutual TLS client authentication programmatically
+
+The `io.quarkus.vertx.http.security.HttpSecurity` CDI event allows to configure the mutual TLS client authentication mechanism programmatically.
+Let's consider the example explained in the <<map-certificate-attributes-to-roles>> section:
+
+[source,properties]
+----
+quarkus.tls.tls-config-1.key-store.p12.path=server-keystore.p12
+quarkus.tls.tls-config-1.key-store.p12.password=the_key_store_secret
+quarkus.tls.tls-config-1.trust-store.p12.path=server-truststore.p12
+quarkus.tls.tls-config-1.trust-store.p12.password=the_trust_store_secret
+quarkus.http.ssl.client-auth=required
+quarkus.http.insecure-requests=disabled
+
+quarkus.http.auth.certificate-role-properties=cert-role-mappings.properties
+
+quarkus.http.auth.permission.certauthenticated.paths=/*
+quarkus.http.auth.permission.certauthenticated.policy=role-policy-cert
+quarkus.http.auth.policy.role-policy-cert.roles-allowed=user,admin
+----
+
+The configuration with the programmatic set up would look like this:
+
+[source,properties]
+----
+quarkus.http.tls-configuration-name=tls-config-1
+----
+
+[source,java]
+----
+package org.acme.http.security;
+
+import io.quarkus.vertx.http.security.HttpSecurity;
+import io.quarkus.tls.BaseTlsConfiguration;
+import io.quarkus.tls.TlsConfigurationRegistry;
+import io.vertx.core.net.KeyCertOptions;
+import io.vertx.core.net.KeyStoreOptions;
+import io.vertx.core.net.SSLOptions;
+import io.vertx.core.net.TrustOptions;
+import jakarta.enterprise.event.Observes;
+
+public class MutualTlsClientAuthConfig {
+
+    void configure(@Observes HttpSecurity httpSecurity, TlsConfigurationRegistry tlsConfigRegistry) {
+        httpSecurity.mechanism(MTLS.builder()
+                .rolesMapping("CN-value", "user", "admin")
+                .build())
+                .path("/*").roles("user", "admin");
+
+        // please note that you can configure your TLS configuration in the 'application.properties' file instead
+        // if and only if you have a good reasons to register this config programmatically, you can do:
```

Comment:
IMHO we should add a `<1>` point here and clarify that you can set in properties, but not discourage it as this comment does; this example gives complete programmatic setup option, which is what the whole idea of the fluent api is :-)